### PR TITLE
Q-IMPL-CONSENSUS-OVERFLOW-DIAGNOSTICS-PARITY-01 (#1189): overflow diagnostics parity

### PR DESCRIPTION
**Pull request by 2tbmz9y2xt-lang**
Scope:
- align Go/Rust overflow diagnostics labels for block resource accumulators (`sum_weight`, `sum_da`, `sum_anchor`)
- add regression tests for named overflow errors in Go and Rust
- fix CLI `runRawJSON` stdin pipe ordering to prevent deep-gate deadlock

Refs: #1189

### CODE REVIEW

File: clients/go/cmd/rubin-consensus-cli/runtime_test.go [hunk 1]
```diff
@@ -154,13 +150,28 @@ func runRawJSON(t *testing.T, raw []byte, entry func()) Response {
 	oldOut := os.Stdout
 	os.Stdin = rIn
 	os.Stdout = wOut
+	defer func() {
+		os.Stdin = oldIn
+		os.Stdout = oldOut
+		_ = rIn.Close()
+		_ = rOut.Close()
```
[the remark this answers is not in the document]
Comment (2tbmz9y2xt-lang) [reply clients/go/cmd/rubin-consensus-cli/runtime_test.go:157]: Fixed in a3ea50b: added deferred `wOut.Close()` in `runRawJSON` cleanup so early exits cannot leave the stdout write end open.
